python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -542,6 +542,41 @@ def wan_ha_interfaces(self: SharedUtils) -> list:
         # Using node values
         return natural_sort(set(self.configured_wan_ha_interfaces), "name")
 
+    @cached_property
+    def wan_ha_port_channel_id(self: SharedUtils) -> int:
+        """Port-channel ID to use for direct WAN HA port-channel.
+
+        If not provided, computed from the list of configured members.
+        """
+        if self.use_uplinks_for_wan_ha:
+            msg = "Using Port-Channel when using uplinks as HA is not supported."
```

Comment:
when would this be raised? This function expects wan_ha_interfaces to be a non-empty list, and will fail badly otherwise.

Reply:
This would be raised if someone tries to use the function in a bad place. I have changed the logic to cater for the "fail badly otherwise"

Reply:
remove the guard